analysis_options.yaml

```diff
@@ -1,15 +1,5 @@
-include: package:pedantic/analysis_options.yaml
+include: package:lints/recommended.yaml
 
 analyzer:
-  strong-mode:
-    implicit-casts: false
```

Comment:
I must object to the naming, though. It should be a lint, not a "strong mode" (which isn't a thing any more) option.

I also occasionally use implicit casts from dynamic *explicitly* because `int x = foo as dynamic;` is cheaper in dart2js than `var  x = foo as int;`.

Reply:
> I must object to the naming, though. It should be a lint, not a "strong mode" (which isn't a thing any more) option.

Yeah I would be in favor of a lint implementing this behavior. If we did that we could have a discussion about whether it should be included in the recommended lint set. cc @bwilkerson 

> I also occasionally use implicit casts from dynamic _explicitly_ because `int x = foo as dynamic;` is cheaper in dart2js than `var x = foo as int;`.

IMO if we want to optimize at that level we'd be better off with a different mechanism like an explicit `unsafeCast` that dart2js recognizes than using a code pattern which hides the trickiness.

Reply:
The last I heard, @srawlins and @leafpetersen had a plan for replacing `strong-mode` options with a set of `language` options. I believe that effort might be stalled at the moment. I don't know whether they considered making these lints and if so why that wasn't the plan. (The code and recommended lint sets didn't exist at the time that effort started, so that might have been a factor.)

Reply:
I'm not sure what "language options" means (probably that you can enable them with command line flags, and they work in the compilers as well as the analyzer). I'd still prefer to think of it as lints, just ones enforceable by the compilers as well.
But that's semantics, and not the programming language kind :)

Reply:
The proposal (the last time I saw it) was to replace `strong-mode` with `language` and to rework the set of options to make them more appropriate to the Dart 2 world (and presumably to the post null-safety world if work on the proposal ever resumes).

If we want any tool other than the analyzer to be able to enforce these options then the analysis options file might not be the right place for it (and making them lints seems even worse).